app/models/one_login/did_cache.rb

```diff
@@ -0,0 +1,102 @@
+require "json"
+require "net/http"
+
+class OneLogin::DidCache
+  # public api to use document
+  # if document not present it will fetch
+  # if it determines document has expired it will first refresh the document
+  def self.document
+    @cache ||= new
+
+    @cache.document
```

Comment:
Should this throw an error if the `@cache.document` is `nil`?

Reply:
it'll probably surface as some other error at the moment as it errors elsewhere. there is a user journey where if theres an error we render a certain page. so can probably do that then